linkerd/distribute/src/keys.rs

```diff
@@ -0,0 +1,160 @@
+use rand::{
+    distributions::{WeightedError, WeightedIndex},
+    prelude::Distribution as _,
+    Rng,
+};
+use std::{collections::HashMap, hash::Hash};
+
+/// Uniquely identifies a key/backend pair for a distribution. This allows
+/// backends to have the same key and still participate in request distribution.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct KeyId {
```

Comment:
It looks like this type scope can be limited--we can keep this an internal implementation detail:
```suggestion
pub(crate) struct KeyId {
```


With this change, we would typically move the declaration down below the real pub types.

Reply:
Done